projects/demo/src/app/app.component.ts

```diff
@@ -8,6 +11,26 @@ import {PermissionsService} from '@ng-web-apis/permissions';
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class AppComponent {
+    error$ = new BehaviorSubject<string | null>(null);
+    basicSample = SAMPLE;
+
+    geolocationState$ = this.permissionsService.state('geolocation').pipe(
+        tap(state => {
+            if (state === 'prompt') {
+                alert(
+                    'After you acknowledge this message the browser will ask your permission to allow location access!',
+                );
+                // calling getCurrentPosition() results in a browser asking for permission if the current status is "prompt"
+                navigator.geolocation.getCurrentPosition(() => {});
```

Comment:
Maybe use the NAVIGATOR token  here? https://github.com/ng-web-apis/common/blob/master/projects/common/src/tokens/navigator.ts

Reply:
sounds good. Will update